sql/rowexec/rel.go

```diff
@@ -532,6 +535,27 @@ func (b *BaseBuilder) populateMax1Results(ctx *sql.Context, n *plan.Max1Row, row
 	return nil
 }
 
+func validateOutfile(dir interface{}, fileStr string) error {
+	if dir == nil || dir == "" {
+		return nil
+	}
+	if _, relErr := filepath.Rel(dir.(string), fileStr); relErr != nil {
+		return sql.ErrSecureFilePriv.New()
+	}
```

Comment:
This is a different check than we do on the load data infile parameter. In particular, the check we preform there has the following behavior differences:

1) It doesn't allow creating files in subdirectories of `secure_file_priv`
2) It uses `os.Open` + `Stat()` on the `filepath.Dir()` of the provided path, in addition to a Stat of the `secure_file_priv` directory, to assert that the files are in the same directory.

The end result is a difference in behavior when there are subdirectories, symlinks, etc. We should probably stick to the same behavior as the read paths?